src/js/player.js

```diff
@@ -2062,6 +2070,26 @@ class Player extends Component {
   }
 
   /**
+   * Returns the current source objects.
+   *
+   * @return {Object[]} The current source objects
+   * @method currentSources
+   */
+  currentSources() {
+    return this.cache_.sources || [this.currentSource()];
+  }
+
+  /**
+   * Returns the current source object.
+   *
+   * @return {Object} The current source object
+   * @method currentSource
+   */
+  currentSource() {
+    return this.cache_.source || { src: this.currentSrc() };
```

Comment:
My initial thought is to throw away all the code for `currentSrc()` since now is very duplicative and loses information such as type. Though this is a breaking change and there is probably a big hole in my logic.

Atm, `currentSrc()` returns a `string` or `undefined`…

I think it makes sense to translate that across those methods. 

```js
currentSrc() === undefined
currentSource() === undefined
currentSources() === undefined

vs.

currentSrc() === undefined
currentSource() === { src: undefined } or {}
currentSources() === [{ src: undefined }] or []
```

So I would recommend `undefined` or `{}` and `[]`.